tests/unit/processing/test_handle_error.py

```diff
@@ -381,35 +381,54 @@ def mock_env_vars():
     }
 
 
-def test_cleanup_artifacts__OK(mock_env_vars):
-    dataset_id = "example_dataset"
-
-    with patch(f"{module_path}.delete_many_from_s3") as mock_delete_many_from_s3, patch.dict(os.environ, mock_env_vars):
-        cleanup_artifacts(dataset_id)
-
-        # Assertions
-        mock_delete_many_from_s3.assert_any_call(os.environ["ARTIFACT_BUCKET"], dataset_id + "/")
-        mock_delete_many_from_s3.assert_any_call(os.environ["DATASETS_BUCKET"], dataset_id + ".")
-        mock_delete_many_from_s3.assert_any_call(os.environ["CELLXGENE_BUCKET"], dataset_id + ".cxg/")
-        assert mock_delete_many_from_s3.call_count == 3
-
-
-def test_cleanup_artifacts__download_validate_no_bucket(caplog):
-    dataset_id = "example_dataset"
-
-    with patch(f"{module_path}.delete_many_from_s3") as mock_delete_many_from_s3, patch.dict(os.environ, clear=True):
-        cleanup_artifacts(dataset_id)
-
-        # Assertions
-        mock_delete_many_from_s3.assert_not_called()
-        assert "Failed to clean up artifacts." in caplog.text
-
-
-def test_cleanup_artifacts__cxg_specified_bucket(mock_env_vars):
-    dataset_id = "example_dataset"
-
-    with patch(f"{module_path}.delete_many_from_s3") as mock_delete_many_from_s3, patch.dict(os.environ, mock_env_vars):
-        cleanup_artifacts(dataset_id)
-
-        # Assertions
-        mock_delete_many_from_s3.assert_called_with(os.environ["CELLXGENE_BUCKET"], dataset_id + ".cxg/")
+class TestCleanupArtifacts:
+    @pytest.mark.parametrize("error_step", ["download-validate", "", None])
+    def test_cleanup_artifacts__OK(self, mock_env_vars, error_step):
+        dataset_id = "example_dataset"
+
+        with patch(f"{module_path}.delete_many_from_s3") as mock_delete_many_from_s3, patch.dict(
+            os.environ, mock_env_vars
+        ):
+            cleanup_artifacts(dataset_id, error_step)
+
+            # Assertions
+            mock_delete_many_from_s3.assert_any_call(os.environ["ARTIFACT_BUCKET"], dataset_id + "/")
+            mock_delete_many_from_s3.assert_any_call(os.environ["DATASETS_BUCKET"], dataset_id + ".")
+            mock_delete_many_from_s3.assert_any_call(os.environ["CELLXGENE_BUCKET"], dataset_id + ".cxg/")
```

Comment:
if you're testing for the call happening with the value taken from `os.environ`, then why are you bothering to patch `os.environ`?